pkg/lwip/0001-Fix-warnings.patch

```diff
@@ -0,0 +1,29 @@
+From 23c41e698bbd97b456ea8502b34bd17f50f3dbea Mon Sep 17 00:00:00 2001
+From: Martine Lenders <mail@martine-lenders.eu>
+Date: Thu, 12 Nov 2015 16:36:00 +0100
+Subject: [PATCH 1/2] Fix warnings
+
+---
+ src/include/lwip/debug.h | 6 ++++--
+ 1 file changed, 4 insertions(+), 2 deletions(-)
+
+diff --git a/src/include/lwip/debug.h b/src/include/lwip/debug.h
+index 973a633..c7b3c9c 100644
+--- a/src/include/lwip/debug.h
++++ b/src/include/lwip/debug.h
+@@ -67,8 +67,10 @@
+  * -- To disable assertions define LWIP_NOASSERT in arch/cc.h.
+  */
+ #ifndef LWIP_NOASSERT
+-#define LWIP_ASSERT(message, assertion) do { if(!(assertion)) \
+-  LWIP_PLATFORM_ASSERT(message); } while(0)
++#define LWIP_ASSERT(message, assertion) \
++    if(!(assertion)) { \
++        LWIP_PLATFORM_ASSERT(message); \
++    }
```

Comment:
The problem is that it is technically perfectly fine code, since the `#ifndef` throws an error anyway if `LWIP_PLATFORM_ASSERT` isn't defined and it's just our `-Werror` that makes problems here. In my experience with the community there I don't think this will come through.